### PR TITLE
OSD-7455 - Adding a Cloud-Ingress Finalizer to IngressController when…

### PR DESCRIPTION
**Pull request by bng0y**
… cleaning up

### CODE REVIEW

**Comment by cblecker** [comment]
/hold
Don't merge until after #188

**Review by sam-nguyen7 — COMMENTED** [review]
lgtm

File: pkg/controller/publishingstrategy/publishingstrategy_controller_test.go [hunk 1]
```diff
@@ -375,3 +387,166 @@ func TestValidatePatchableStatus(t *testing.T) {
...
+func getK8sError(errorType string, obj client.Object) error {
+	errorMap := map[string]error{
+		"IsNotFound": k8serr.NewNotFound(schema.GroupResource{Group: "ingresscontrollers.cloudingress.managed.openshift.io",
+			Resource: "IngressController"}, obj.GetName()),
+	}
+	if err, found := errorMap[errorType]; found {
+		return err
+	} else {
+		// by default we return internal error, when the error type specified doesn't match something we preconfigured
+		return k8serr.NewInternalError(errors.New(fmt.Sprintf("%v was raised", errorType)))
```
[the remark this answers is not in the document]
Comment (bng0y) [reply pkg/controller/publishingstrategy/publishingstrategy_controller_test.go]: solved. thanks

File: pkg/controller/publishingstrategy/publishingstrategy_controller.go [hunk 2]
```diff
@@ -528,3 +540,51 @@ func getIngressWithCloudIngressOpreatorOwnerAnnotation(ingressList operatorv1.In
...
+
+func (r *ReconcilePublishingStrategy) ensureIngressController(reqLogger logr.Logger, ingressController, desiredIngressController *operatorv1.IngressController) (reconcile.Result, error) {
+	// If ingresscontroller still has the ClusterIngressFinalizer, there is no point continuing.
+	// Cluster-ingress-operator typically needs a few minutes to delete all dependencies
+	if ctlutils.Contains(ingressController.GetFinalizers(), ClusterIngressFinalizer) {
+		reqLogger.Info(fmt.Sprintf("%s IngressController's  is in the process of being deleted, requeing", ingressController.Name))
+		return reconcile.Result{Requeue: true, RequeueAfter: 30 * time.Second}, nil
+	}
+
+	// At this point, ClusterIngressFinalizer is gone, meaning cluster-ingerss-operator has completed dependency cleanup
```
[the remark this answers is not in the document]
Comment (bng0y) [reply pkg/controller/publishingstrategy/publishingstrategy_controller.go]: fixed. thanks

**Pull request merged**